### PR TITLE
TELCODOCS-1684 - RAN and Core 4.17 RDS updates

### PR DESCRIPTION
**Pull request by aireilly**
RAN and Core 4.17 RDS updates

Version(s):
enterprise-4.17

Issue:
https://issues.redhat.com/browse/TELCODOCS-1684

Link to docs preview:
* [RAN RDS](https://81822--ocpdocs-pr.netlify.app/openshift-enterprise/latest/scalability_and_performance/telco_ref_design_specs/ran/telco-ran-ref-design-spec)
* [Core RDS](https://81822--ocpdocs-pr.netlify.app/openshift-enterprise/latest/scalability_and_performance/telco_ref_design_specs/core/telco-core-rds-overview)

QE review:
- [x] QE has approved this change.

### CODE REVIEW

File: modules/telco-ran-lvms-operator.adoc [hunk 1]
```diff
@@ -44,7 +46,11 @@ spec:
...
 
 Limits and requirements::
-* In {sno} clusters, persistent storage must be provided by either LVMS or local storage, not both.
+* In {sno} clusters, persistent storage must be provided by either {lvms} or local storage, not both.
+* Volume snapshots are excluded from the reference configuration.
 
 Engineering considerations::
+* {lvms} can be used as the local storage implementation for the RAN DU use case.
+When {lvms} is used as the storage solution, it replaces the Local Storage Operator, and the CPU required is assigned to the management partition as platform overhead.
+The reference configuration must include one of these storage solutions but not both.
```
Comment (lpettyjo) [inline modules/telco-ran-lvms-operator.adoc:55]: Add comma: "...orage solutions, but not both."

**Review by lpettyjo — COMMENTED** [review]
You need to squash your three commit into one. For info, see https://docs.google.com/document/d/1fLMpK4bqthtFlCwA36yeo2cI-pExFpX76VMnSi_In6Q/edit#heading=h.epes4xf06pzk, and let me know if you have any questions.

Otherwise, LGTM!

File: modules/telco-ran-crs-machine-configuration.adoc [hunk 2]
```diff
@@ -12,18 +12,18 @@
...
+Kdump enable,xref:../../telco_ref_design_specs/ran/telco-ran-ref-du-crs.adoc#ztp-06-kdump-master-yaml[06-kdump-master.yaml],No,No
+Kdump enable,xref:../../telco_ref_design_specs/ran/telco-ran-ref-du-crs.adoc#ztp-06-kdump-worker-yaml[06-kdump-worker.yaml],No,No
+Kubelet configuration / Container mount hiding,xref:../../telco_ref_design_specs/ran/telco-ran-ref-du-crs.adoc#ztp-01-container-mount-ns-and-kubelet-conf-master-yaml[01-container-mount-ns-and-kubelet-conf-master.yaml],No,No
+Kubelet configuration / Container mount hiding,xref:../../telco_ref_design_specs/ran/telco-ran-ref-du-crs.adoc#ztp-01-container-mount-ns-and-kubelet-conf-worker-yaml[01-container-mount-ns-and-kubelet-conf-worker.yaml],No,No
 One-shot time sync,xref:../../telco_ref_design_specs/ran/telco-ran-ref-du-crs.adoc#ztp-99-sync-time-once-master-yaml[99-sync-time-once-master.yaml],No,No
 One-shot time sync,xref:../../telco_ref_design_specs/ran/telco-ran-ref-du-crs.adoc#ztp-99-sync-time-once-worker-yaml[99-sync-time-once-worker.yaml],No,No
 SCTP,xref:../../telco_ref_design_specs/ran/telco-ran-ref-du-crs.adoc#ztp-03-sctp-machine-config-master-yaml[03-sctp-machine-config-master.yaml],No,No
```
Comment (imiller0) [inline modules/telco-ran-crs-machine-configuration.adoc:23]: SCTP is optional

File: modules/telco-ran-gitops-operator-and-ztp-plugins.adoc [hunk 3]
```diff
@@ -12,7 +13,10 @@ New in this release::
 Description::
 {gitops-shortname} and {ztp} plugins provide a {gitops-shortname}-based infrastructure for managing cluster deployment and configuration.
 Cluster definitions and configurations are maintained as a declarative state in Git.
-ZTP plugins provide support for generating installation CRs from the `SiteConfig` CR and automatic wrapping of configuration CRs in policies based on `PolicyGenTemplate` CRs.
+You can apply `ClusterInstance` CRs to the hub cluster where the `SiteConfig` Operator renders them as installation CRs.
+Alternatively, you can use the {ztp} plugin to generate installation CRs directly from `SiteConfig` CRs.
+The {ztp} plugin supports automatic wrapping of `Configuration` CRs in policies based on `PolicyGenTemplate` CRs.
```
Comment (imiller0) [inline modules/telco-ran-gitops-operator-and-ztp-plugins.adoc:18]: In this case Configuration should be all lowercase (it isn't a type but rather a category of crs)

File: modules/telco-ran-gitops-operator-and-ztp-plugins.adoc [hunk 4]
```diff
@@ -12,7 +13,10 @@ New in this release::
...
+
 +
 You can deploy and manage multiple versions of {product-title} on managed clusters using the baseline reference configuration CRs.
```
Comment (imiller0) [inline modules/telco-ran-gitops-operator-and-ztp-plugins.adoc:21]: Can we augment this to include a couple things
- Supporting multiple versions is possible when the source CRs and policy (PolicyGenTemplate or PolicyGenerator) are all stored in git, allowing creation of per-version policies simultaneously
- We recommend keeping reference CRs and custom CRs under different directories. This allows a patch/update of the reference CRs by simple replacement of all directory contents without disturbing the custom CRs content.

File: modules/ztp-telco-ran-software-versions.adoc [hunk 5]
```diff
@@ -16,25 +16,25 @@ The Red Hat telco RAN DU {product-version} solution has been validated using the
...
 
-|Node Tuning Operator
-|4.16
+|PTP Operator
+|4.17
 
-|Logging Operator
-|4.16
+|SRIOV Operator
+|4.17
```
Comment (imiller0) [inline modules/ztp-telco-ran-software-versions.adoc:37]: Need to add the Life Cycle Agent operator

**Review by imiller0 — COMMENTED** [review]
Great work. Some comments/edits noted.

File: snippets/telco-core_ClusterLogSubscription.yaml [hunk 6]
```diff
@@ -4,7 +4,7 @@ metadata:
   name: cluster-logging
   namespace: openshift-logging
 spec:
-  channel: "stable"
+  channel: "stable-6.0"
   name: cluster-logging
   source: redhat-operators-disconnected
   sourceNamespace: openshift-marketplace
```
Comment (yprokule) [inline snippets/telco-core_ClusterLogSubscription.yaml:10]: worth also update `installPlanApproval` to `Manual`

File: snippets/ztp_example-sno.yaml [hunk 7]
```diff
@@ -72,8 +72,8 @@ spec:
           bmcCredentialsName:
             name: "example-node1-bmh-secret"
           bootMACAddress: "AA:BB:CC:DD:EE:11"
-          # Use UEFISecureBoot to enable secure boot
-          bootMode: "UEFI"
+          # Use UEFISecureBoot to enable secure boot, UEFI to disable.
```
Comment (yprokule) [inline snippets/ztp_example-sno.yaml:75]: We need to confirm this part, iirc setting `UEFI` mode doesn't disable previously/already configured _SecureBoot_  so this should be done manually

File: modules/telco-ran-ptp-operator.adoc [hunk 8]
```diff
@@ -1,45 +1,40 @@
...
-
-* You can configure the `linuxptp` services `ptp4l` and `phc2sys` as a highly available (HA) system clock for dual PTP boundary clocks (T-BC).
+* A new version two of the Precision Time Protocol (PTP) fast event REST API is available.
+Consumer applications can now subscribe directly to the events REST API in the PTP events producer sidecar.
+The PTP fast event REST API v2 is compliant with the link:https://orandownloadsweb.azurewebsites.net/download?id=344[O-RAN O-Cloud Notification API Specification for Event Consumers 3.0].
+You can change the API version by setting the `ptpEventConfig.apiVersion` field in the `PtpOperatorConfig` resource.
 
 Description::
-See link:https://docs.openshift.com/container-platform/4.17/scalability_and_performance/ztp_far_edge/ztp-reference-cluster-configuration-for-vdu.html#ztp-sno-du-configuring-ptp_sno-configure-for-vdu[PTP timing] for details of support and configuration of PTP in cluster nodes.
+See "Recommended {sno} cluster configuration for vDU application workloads" for details of support and configuration of PTP in cluster nodes.
```
[the remark this answers is not in the document]
Comment (aireilly) [reply modules/telco-ran-ptp-operator.adoc:16]: Discussed this with @sabbir-47 just now. I will create a JIRA to track this.

[the remark this answers is not in the document]
Comment (aireilly) [reply modules/telco-ran-ptp-operator.adoc:16]: https://issues.redhat.com/browse/TELCODOCS-2054

File: snippets/ztp_example-sno.yaml (hunk 7 above)
Comment (aireilly) [reply snippets/ztp_example-sno.yaml:75]: @imiller0 can you comment?

**Comment by kalexand-rh** [comment]
The `branch/enterprise-4.18` label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.17. And any PR going into `main` must also target the latest version branch (enterprise-4.18).

If the update in your PR does NOT apply to version 4.18 onward, please re-target this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main.

File: snippets/telco-core_ClusterLogSubscription.yaml (hunk 6 above)
Comment (yprokule) [reply snippets/telco-core_ClusterLogSubscription.yaml:10]: It should be there(https://github.com/openshift-kni/telco-reference/pull/32), but let's wait for the containers to pick up the change, thanks

**Comment by imiller0** [comment]
The RAN RDS updates lgtm

**Comment by MarSik** [comment]
As agreed, this looks good to me from the Compute and Core perspective assuming Ian will double check the ACM related pieces.

/lgtm

File: modules/telco-core-cpu-partitioning-performance-tune.adoc [hunk 9]
```diff
@@ -42,4 +42,6 @@ Engineering considerations::
 * Hardware without Interrupt Request (IRQ) affinity support will impact isolated CPUs. To ensure that pods with guaranteed CPU QoS have full use of allocated CPU, all hardware in the server must support IRQ affinity.
 * OVS dynamically manages its `cpuset` configuration to adapt to network traffic needs.
 You do not need to reserve additional CPUs for handling high network throughput on the primary CNI.
-* If workloads running on the cluster require cgroups v1, you can configure nodes to use cgroups v1. You can make this configuration as part of initial cluster deployment. For more information, see _Enabling Linux cgroup v1 during installation_ in the _Additional resources_ section.
+* If workloads running on the cluster require cgroups v1, you can configure nodes to use cgroups v1.
+You make this configuration as part of initial cluster deployment.
```
Comment (MarSik) [inline modules/telco-core-cpu-partitioning-performance-tune.adoc:46]: Aidan: here the "You can make" vs "You make"

File: snippets/ztp_example-sno.yaml (hunk 7 above)
Comment (imiller0) [reply snippets/ztp_example-sno.yaml:75]: Agreed. Poorly worded comment. I believe that UEFI simply leaves secureboot "as is", whereas "UEFISecureBoot" enables it. I will revert that comment to "# Use UEFISecureBoot to enable secure boot"

File: modules/ztp-telco-ran-software-versions.adoc [hunk 10]
```diff
@@ -16,28 +16,28 @@ The Red Hat telco RAN DU {product-version} solution has been validated using the
 |Software version
 
 |Managed cluster version
-|4.16
+|4.17
```
Comment (stevsmit) [inline modules/ztp-telco-ran-software-versions.adoc:19]: Same thought here about the attribute as above.

File: scalability_and_performance/telco_ref_design_specs/core/telco-core-ref-design-components.adoc [hunk 11]
```diff
@@ -35,28 +34,30 @@ include::modules/telco-core-rds-networking.adoc[leveloffset=+1]
...
-include::modules/telco-core-sriov.adoc[leveloffset=+2]
+include::modules/telco-core-sriov.adoc[leveloffset=+1]
 
 [role="_additional-resources"]
 .Additional resources
 
 * xref:../../../networking/hardware_networks/about-sriov.adoc#about-sriov[About Single Root I/O Virtualization (SR-IOV) hardware networks]
 
-include::modules/telco-nmstate-operator.adoc[leveloffset=+2]
+* link:../../../networking/hardware_networks/about-sriov.adoc#supported-devices_about-sriov[Supported devices]
```
Comment (stevsmit) [inline scalability_and_performance/telco_ref_design_specs/core/telco-core-ref-design-components.adoc:58]: Page not found with this xref.

File: scalability_and_performance/telco_ref_design_specs/core/telco-core-ref-design-components.adoc [hunk 12]
```diff
@@ -75,6 +76,10 @@ include::modules/telco-core-power-management.adoc[leveloffset=+1]
 [role="_additional-resources"]
 .Additional resources
 
+* xref:../../../rest_api/node_apis/performanceprofile-performance-openshift-io-v2.adoc#spec-workloadhints[Performance Profile]
+
+* link:../../../scalability_and_performance/low_latency_tuning/cnf-tuning-low-latency-nodes-with-perf-profile.adoc#cnf-configuring-power-saving-for-nodes_cnf-low-latency-perf-profile[Configuring power saving for nodes]
```
Comment (stevsmit) [inline scalability_and_performance/telco_ref_design_specs/core/telco-core-ref-design-components.adoc:81]: Page not found with this xref.

File: scalability_and_performance/telco_ref_design_specs/ran/telco-ran-ref-du-components.adoc [hunk 13]
```diff
@@ -11,49 +11,107 @@ The following sections describe the various {product-title} components and confi
...
 
-* xref:../../../networking/hardware_networks/configuring-sriov-qinq-support.adoc[Configuring QinQ support for SR-IOV enabled workloads]
+* xref:../../../networking/hardware_networks/configuring-sriov-qinq-support.adoc#configuring-qinq-support[Configuring QinQ support for SR-IOV enabled workloads]
 
 include::modules/telco-ran-logging.adoc[leveloffset=+1]
 
+[role="_additional-resources"]
+.Additional resources
+
+* xref:../../../observability/logging/cluster-logging.adoc#cluster-logging[Logging]
```
Comment (stevsmit) [inline scalability_and_performance/telco_ref_design_specs/ran/telco-ran-ref-du-components.adoc:51]: Technically links to "About Logging"

**Review by stevsmit — COMMENTED** [review]
Not a lot of comments (I can see this PR has been thoroughly reviewed probably multiple times). Biggest concerns are there's module that isn't included in an assembly that has changes, and a few of the xrefs are broken. There's some small things like adding {nbsp} between Red and Hat if you want to fix those, but other than that the content looks really good. 

Took forever to review it; I tried to be thorough and look at the preview and the diff.

File: modules/telco-core-software-stack.adoc [hunk 14]
```diff
@@ -1,28 +1,33 @@
...
+|====
 |Component |Software version
 
-|Cluster Logging Operator |5.9.1
+|Cluster Logging Operator
+|6.0
 
-|{rh-storage} |4.16
+|{rh-storage}
+|4.17
```
[the remark this answers is not in the document]
Comment (aireilly) [reply modules/telco-core-software-stack.adoc:20]: Not always. We need to hard code these.

File: modules/telco-core-agent-based-installer-abi.adoc [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+
+:_mod-docs-content-type: REFERENCE
+[id="telco-core-agent-based-installer-abi_{context}"]
+= Agent-based installer
+
+New in this release::
+* No reference design updates in this release
+
+Description::
+You can install telco core clusters with the Agent-based installer (ABI) on bare-metal servers without requiring additional servers or virtual machines for managing the installation.
```
[the remark this answers is not in the document]
Comment (rohennes) [reply modules/telco-core-agent-based-installer-abi.adoc:13]: I know we had this conversation previously regarding bare-metal vs bare metal. Currently "bare metal" would flag a Vale error, resulting in a Vale bot comment also, so for now I'd like to keep it as bare-metal. 
(Vale [rule](https://github.com/redhat-documentation/vale-at-red-hat/blob/main/.vale/styles/RedHat/TermsErrors.yml#L23))

File: modules/telco-core-installation.adoc [hunk 16]
```diff
@@ -1,28 +1,19 @@
 // Module included in the following assemblies:
 //
-// * telco_ref_design_specs/ran/telco-core-ref-components.adoc
+// * scalability_and_performance/telco_ref_design_specs/core/telco-core-ref-design-components.adoc
 
 :_mod-docs-content-type: REFERENCE
 [id="telco-core-installation_{context}"]
 = Installation
```
[the remark this answers is not in the document]
Comment (rohennes) [reply modules/telco-core-installation.adoc:7]: Well spotted" I believe this is OK though. There is now a deployment components section and ABI is covered there instead: https://81822--ocpdocs-pr.netlify.app/openshift-enterprise/latest/scalability_and_performance/telco_ref_design_specs/core/telco-core-ref-design-components#telco-core-agent-based-installer-abi_core-ref-design-components

File: modules/telco-core-load-balancer.adoc [hunk 17]
```diff
@@ -1,20 +1,27 @@
...
-* No reference design updates in this release
+//CNF-11914
+* In {product-title} 4.17, `frr-k8s` is now the default and fully supported Border Gateway Protocol (BGP) backend.
+The deprecated `frr` BGP mode is still available.
+You should upgrade clusters to use the `frr-k8s` backend.
 
 Description::
-
-MetalLB is a load-balancer implementation for bare metal Kubernetes clusters using standard routing protocols. It enables a Kubernetes service to get an external IP address which is also added to the host network for the cluster.
+MetalLB is a load-balancer implementation that uses standard routing protocols for bare-metal clusters. It enables a Kubernetes service to get an external IP address which is also added to the host network for the cluster.
```
[the remark this answers is not in the document]
Comment (rohennes) [reply modules/telco-core-load-balancer.adoc:16]: See previous comment re bare metal.

File: modules/telco-core-rds-networking.adoc [hunk 18]
```diff
@@ -1,10 +1,38 @@
...
+* Bond and VLAN interfaces are created at install time during network configuration.
+Apart from the VLAN (`VLAN0`) used by the primary CNI, the other VLANS can be created on Day 2 using the Kubernetes NMState Operator.
+
+* MACVLAN and IPVLAN interfaces are created with their corresponding CNIs.
+They do not share the same base interface.
+
+* SR-IOV VFs are managed by the SR-IOV Network Operator.
+The following diagram provides an overview of SR-IOV NIC sharing:
++
+.SR-IOV NIC sharing
```
[the remark this answers is not in the document]
Comment (rohennes) [reply modules/telco-core-rds-networking.adoc:37]: True... perhaps has something to do with the plus continuation? I don't think it's a show-stopper though. I'm happy to give this a pass considering the urgency of getting this merged.

File: modules/telco-core-red-hat-advanced-cluster-management.adoc [hunk 19]
```diff
@@ -0,0 +1,15 @@
+// Module included in the following assemblies:
+//
+// * scalability_and_performance/telco_ref_design_specs/core/telco-core-ref-design-components.adoc
+
+:_mod-docs-content-type: REFERENCE
+[id="telco-core-red-hat-advanced-cluster-management_{context}"]
```
[the remark this answers is not in the document]
Comment (rohennes) [reply modules/telco-core-red-hat-advanced-cluster-management.adoc:6]: Yeap looks like it, deleted, thanks!

File: modules/telco-ran-ptp-operator.adoc [hunk 20]
```diff
@@ -1,45 +1,40 @@
...
 +
 * As an ordinary clock (OC) synced to a grandmaster clock or boundary clock (T-BC)
 
-* As a grandmaster clock synced from GPS with support for single or dual card E810 Westport Channel NICs.
-
-* As dual boundary clocks (one per NIC) with support for E810 Westport Channel NICs
+* As a grandmaster clock (T-GM) synced from GPS with support for single or dual card E810 NICs.
 
-* Allow for High Availability of the system clock when there are multiple time sources on different NICs.
+* As dual boundary clocks (one per NIC) with support for E810 NICs
```
[the remark this answers is not in the document]
Comment (rohennes) [reply modules/telco-ran-ptp-operator.adoc:23]: Fixed

File: modules/ztp-telco-ran-software-versions.adoc (hunk 10 above)
Comment (rohennes) [reply modules/ztp-telco-ran-software-versions.adoc:19]: See Aidan's previous, need to hardcode these.

File: scalability_and_performance/telco_ref_design_specs/core/telco-core-ref-design-components.adoc (hunk 11 above)
Comment (rohennes) [reply scalability_and_performance/telco_ref_design_specs/core/telco-core-ref-design-components.adoc:58]: This is a link, should be an xref. Updated.

File: scalability_and_performance/telco_ref_design_specs/core/telco-core-ref-design-components.adoc (hunk 12 above)
Comment (rohennes) [reply scalability_and_performance/telco_ref_design_specs/core/telco-core-ref-design-components.adoc:81]: This is a link, should be an xref. Updated.

File: modules/telco-core-rds-networking.adoc (hunk 18 above)
[the remark this answers is not in the document]
Comment (stevsmit) [reply modules/telco-core-rds-networking.adoc:37]: Definitely not. I wouldn't even have called this out in a merge review had there not been a few other things that needed addressed. NP though.

**Pull request merged**